packages/@dynamic-labs-connectors/sequence-cross-app-evm/src/SequenceCrossAppConnector.ts

```diff
@@ -0,0 +1,301 @@
+import { type EthereumWalletConnectorOpts } from "@dynamic-labs/ethereum-core";
+import {
+  EthereumInjectedConnector,
+  type IEthereum,
+} from "@dynamic-labs/ethereum";
+import { DynamicError } from "@dynamic-labs/utils";
+import {
+  WalletMetadata,
+  type Chain,
+} from "@dynamic-labs/wallet-connector-core";
+import { findWalletBookWallet } from "@dynamic-labs/wallet-book";
+import { toHex, getAddress, TransactionRejectedRpcError } from "viem";
+import { ethers } from "ethers";
+import { allNetworks, EIP1193Provider } from "@0xsequence/network";
+import { WalletConnector } from "@dynamic-labs/wallet-connector-core";
+
+import { ProviderTransport } from "./providerTransport";
+
+export const createSequenceCrossAppConnector = (
+  metadata: WalletMetadata,
+  transportConfig: CrossAppTransportConfig
+): [new (props: EthereumWalletConnectorOpts & { transportConfig: CrossAppTransportConfig }) => WalletConnector] => {
+  return [
+    class extends SequenceCrossAppConnector {
+      constructor(
+        props: EthereumWalletConnectorOpts & {
+          transportConfig: CrossAppTransportConfig;
+        }
+      ) {
+        super({
+          ...props,
+          metadata,
+          transportConfig,
+        });
+      }
+    },
+  ];
+};
+
+export type CrossAppTransportConfig = {
+  projectAccessKey: string;
+  walletUrl: string;
+  initialChainId: number;
+};
+
+export class SequenceCrossAppConnector extends EthereumInjectedConnector {
+  private walletName: string;
+
+  private nodesUrl = "https://nodes.sequence.app";
+
+  sequenceWaasTransportProvider: SequenceWaasTransportProvider;
+
+  /**
+   * The name of the wallet connector
+   * @override Required override from the base connector class
+   */
+  override get name() {
+    return this.walletName;
+  }
+
+  /**
+   * The constructor for the connector, with the relevant metadata
+   * @param props The options for the connector
+   */
+  constructor(
+    props: EthereumWalletConnectorOpts & {
+      transportConfig: CrossAppTransportConfig;
+    }
+  ) {
+    super({
+      ...props,
+    });
+
+    this.walletName = props.metadata!.id;
```

Comment:
you need to pass the metadata here. See safe connector for reference:
https://github.com/dynamic-labs/public-wallet-connectors/blob/main/packages/%40dynamic-labs-connectors/safe-evm/src/SafeEvmWalletConnector.ts#L23-L27

Reply:
it is passed in `createSequenceCrossAppConnector`, lmk if that looks ok to you

Reply:
the metadata shouldn't be variable, so please move it to the constructor

Reply:
also, make sure you add the iconUrl